### PR TITLE
Delta Lake docs

### PR DESCRIPTION
**Pull request by acquamarin**

### CODE REVIEW

File: src/content/docs/extensions/delta.mdx
Comment (ray6080) [inline src/content/docs/extensions/delta.mdx:1]: Should we document supported KuzuDB versions?

Comment (ray6080) [inline src/content/docs/extensions/delta.mdx:1]: Please also update `## Available extensions` table in `index.mdx`.

File: src/content/docs/extensions/delta.mdx [hunk 1]
```diff
@@ -129,3 +129,6 @@ RETURN *;
 CREATE NODE TABLE student (name STRING, ID INT64, PRIMARY KEY(ID));
 COPY student FROM 's3://kuzu-sample/student-delta' (file_format='delta')
 ```
+
+### Limitations
```
Comment (prrao87) [inline src/content/docs/extensions/delta.mdx:133]: @acquamarin could you copy the limitations from the Iceberg PR?

File: src/content/docs/extensions/delta.mdx
Comment (prrao87) [reply src/content/docs/extensions/delta.mdx:1]: I don't think so - we don't do this for the httpfs extension as well, so we can leave this part out for now.

**Pull request merged**